__tests__/todo.spec.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { render, fireEvent, cleanup } from '@testing-library/react-native';
+import { build, fake } from '@jackfranklin/test-data-bot';
+import { firebase as mockFirebase } from '../src/firebase/config';
+
+import Todo from '../src/screens/todo';
+
+// mockFirebase.firestore().useEmulator('localhost', 8080);
+
+const userDataBuilder = build('User Data', {
+  fields: {
+    uid: fake((f) => f.internet.password()),
+    fullName: fake((f) => f.name.findName()),
+    email: fake((f) => f.internet.email()),
+  },
+});
+const user = userDataBuilder();
+
+const dataBuilder = build('Data', {
+  fields: {
+    word: fake((f) => f.lorem.word()),
+  },
+});
+const data = dataBuilder();
+
+const addItem = (targetInput, itemText) => {
+  fireEvent.changeText(targetInput, itemText);
+  fireEvent(targetInput, 'submitEditing');
+};
+
+afterEach(() => cleanup());
+
+test('user is able to add an item to the to-do list', async () => {
+  //* render the todo list
+  const { getByText, getByPlaceholderText } = render(<Todo user={user} />);
+
+  //* type in the input field & click submits
+  addItem(getByPlaceholderText('What needs to be done?'), data.word);
+
+  //* todo item should be added to the list
+  expect(getByText(data.word)).toBeDefined();
+  expect(getByText('1 items')).toBeDefined();
+});
+
+test('user should be able to toggle to-do item', async () => {
+  //* render the todo list
+  const { getByPlaceholderText, getByLabelText } = render(<Todo user={user} />);
+
+  //* type in the input field & click submits
+  addItem(getByPlaceholderText('What needs to be done?'), data.word);
+  const itemSwitch = getByLabelText(`${data.word}-switch`);
```

Comment:
`${data.word}-switch` will result in perhaps `hola mundo-switch` id, why not to use the `id` of a to-do instead to composite a the DOM attribute `id`

Reply:
Move to use `getByRole('switch')` cause there is just one item when runnig that test, a that way less refactor on the source
See commit `#adbb9cd` (last commit at the writing of this message)